pkg/common/apimachinery_status_conditions.go

```diff
@@ -16,3 +19,35 @@ func ConditionMarshal(conditions []metav1.Condition) ([]byte, error) {
 	})
 	return json.Marshal(condCopy)
 }
+
+// AcceptedCondition returns an accepted conditions with common reasons for a kuadrant policy
+func AcceptedCondition(policy KuadrantPolicy, err error) *metav1.Condition {
+	// Accepted
+	cond := &metav1.Condition{
+		Type:    string(gatewayapiv1alpha2.PolicyConditionAccepted),
+		Status:  metav1.ConditionTrue,
+		Reason:  string(gatewayapiv1alpha2.PolicyReasonAccepted),
+		Message: fmt.Sprintf("%s has been accepted", policy.Kind()),
+	}
+
+	if err != nil {
+		cond.Status = metav1.ConditionFalse
+		cond.Message = err.Error()
+
+		switch {
+		// TargetNotFound
+		case errors.As(err, &ErrTargetNotFound{}):
```

Comment:
Thinking if this we could avoid this `switch..case` statement. What about defining an interface (in `pkg/common/errors.go`) that each error type would implement?

Reply:
Yup, sounds good to me 👍